### PR TITLE
Forms

### LINKED ISSUE #58: Update registration page
Depends on https://github.com/cfpb/hmda-platform/issues/778

Once https://github.com/cfpb/hmda-platform/issues/778 is complete we'll have a few changes to make based on the new endpoint and its response.

### PR DESCRIPTION
**Pull request by awolfe76**
Closes #58
- removes `autocomplete="off"`, this is a pretty nice feature not sure why we wouldn't keep it
- builds list of checkboxes based on institutions search results
  - includes error messaging for when no results are returned
  - defaults to `checked` if only 1 institution is returned
  - outputs RSSD as the ID ... __should we use another ID or more than one?__

#### Screenshot
register

### CODE REVIEW

File: keycloak/themes/hmda/login/login.ftl [hunk 1]
```diff
@@ -15,7 +15,7 @@
                     <#if usernameEditDisabled??>
                         <input id="username" name="username" type="text" autocapitalize="off" autocorrect="off" value="${(login.username!'')?html}" disabled>
                     <#else>
-                        <input id="username" name="username" value="${(login.username!'')?html}" type="text" autofocus autocomplete="off" />
+                        <input id="username" name="username" value="${(login.username!'')?html}" type="text" autofocus />
```
Comment (hkeeler) [inline keycloak/themes/hmda/login/login.ftl]: I think I'm OK removing `autocomplete=off`.  It does provide some amount of added security, but I would argue probably not enough to warrant the usability ding.  It also sounds like [many browsers don't honor `autocomplete=off` for username fields](https://developer.mozilla.org/en-US/docs/Web/Security/Securing_your_site/Turning_off_form_autocompletion#The_autocomplete_attribute_and_login_fields) anyway.

All that said, don't be surprised if this doesn't come back up during security review.

Comment (awolfe76) [inline keycloak/themes/hmda/login/login.ftl]: Yeah, just seems strange to me on the `username` field ... not totally sure what the security benefits are there, but for UX its very nice to have it on.

**Comment by hkeeler** [comment]
Came across an issue while testing this out locally.  Whenever you get the _Create an account_ screen into its error state (say, password mismatch), the institution is automatically selected, but it is not recognized when you click _Register_.  You get the _Please specify the Institution(s) for which you will be filing HMDA data._ message as if it were not checked.  However, if you uncheck and recheck the box, it does accept the institution.

**Comment by hkeeler** [comment]
Confirmed issue described on https://github.com/cfpb/hmda-platform-auth/pull/67#issuecomment-277807952 is now fixed.

**Comment by hkeeler** [comment]
I like it!  The checkboxes work really well, and seem more stable than the Select2 dropdown widget.  

The code itself looks good.  Just a few thoughts on how all this currently works:

1. I think we should display all of `externalIds`, not just _RSSD ID_.  The question is how do we display it effectively.  In the previous version, each ID was stacked vertically.  Now that it is all inline with the institution name, I think it will probably line wrap when the bank name is long and/or there are 3+ additional IDs.  This will likely be hard to read.

    Perhaps we could do a hybrid of the new and old versions?  Keep the checkbox, but with a layout similar to items in the Select2 dropdown?  I think it will be easier to read, and find the ID you're most familiar with.

1. Let's display what comes in `externalIds.name`.  If we want to change one of them, let's change it on the API so it will cascade down to all other places that'll use it.

1. It seems a little unclear what the user is supposed to do to get the list of institutions to show up.  It says _After entering your email address a list of available institutions will appear._, but that's not exactly what happens.  Am I supposed to click into that area?  I'm guessing most users would figure it out, but I still think it could be a little clearer.

    We _could_ have a greyed out example checkbox and bank name so the user knows what to expect once they fill it in.

Thoughts?

**Comment by awolfe76** [comment]
👍 

For #1, I'll play around a bit but I definitely agree. This could all feel a bit cleaner.

For #2, makes sense.

For #3, I meant to ask if you know how the email validation works on the back-end, or if you could find out? I'd like to use the same check on the front-end too, so that the API call doesn't have to happen "on blur" but could happen during typing. As soon as a valid email address is entered make the API call. This way they would see some results faster, they wouldn't have to leave the email input. I think that will make it feel a bit smoother. But I'd like to use the same regex in both places if we can.

**Comment by hkeeler** [comment]
Here's the regex Keycloak uses for validating email addresses:
* https://github.com/keycloak/keycloak/blob/c6363aa1467a90522f2c8c84dd29f30c3af0e827/services/src/main/java/org/keycloak/services/validation/Validation.java#L45

That said, email validation is a tricky beast.  It will likely start to be _valid_ before it is _complete_.  This will probably mean extra API calls until they fully fill in their address.  Perhaps not a big deal.  Worth some experimentation anyway.

**Comment by awolfe76** [comment]
@hkeeler - just pushed a few updates. Have a look when you get a chance.

#### Screenshot
register

**Comment by hkeeler** [comment]
@awolfe76, looks great!  There's a little bit of tab weirdness when you go from the email field to the institution list, but I don't think we need to hold this PR over that.  I'll create a separate issue.  Merging...

**Pull request merged**